From 647e9a996d49ddc08becf02c734cc37ea05f5286 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Fri, 5 Feb 2016 02:39:21 -0500
Subject: [PATCH] wr-policy: fix avc denials from watchdog

Fix avc denials:

* Patches ported from Fedora 22:
  Reference sources: selinux-policy-3.13.1-128.21.fc22.src.rpm
  Changes come from: policy-f22-contrib.patch

  avc: denied { create } for pid=372 comm="watchdog" name="watchdog" \
  scontext=system_u:system_r:watchdog_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:var_log_t:s0 \
  tclass=dir

* WRL fix:

  avc: denied { search } for pid=385 comm="watchdog" name="journal" \
  dev="tmpfs" ino=8203 \
  scontext=system_u:system_r:watchdog_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:syslogd_var_run_t:s15:c0.c1023 \
  tclass=dir

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/contrib/watchdog.te | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/policy/modules/contrib/watchdog.te b/policy/modules/contrib/watchdog.te
index 0f13e2b..b42f344 100644
--- a/policy/modules/contrib/watchdog.te
+++ b/policy/modules/contrib/watchdog.te
@@ -30,8 +30,9 @@ allow watchdog_t self:fifo_file rw_fifo_file_perms;
 allow watchdog_t self:rawip_socket create_socket_perms;
 allow watchdog_t self:tcp_socket { accept listen };
 
-allow watchdog_t watchdog_log_t:file { append_file_perms create_file_perms setattr_file_perms };
-logging_log_filetrans(watchdog_t, watchdog_log_t, file)
+manage_files_pattern(watchdog_t,watchdog_log_t,watchdog_log_t)
+manage_dirs_pattern(watchdog_t,watchdog_log_t,watchdog_log_t)
+logging_log_filetrans(watchdog_t, watchdog_log_t, {dir file})
 
 manage_files_pattern(watchdog_t, watchdog_var_run_t, watchdog_var_run_t)
 files_pid_filetrans(watchdog_t, watchdog_var_run_t, file)
@@ -99,3 +100,8 @@ optional_policy(`
 optional_policy(`
 	udev_read_db(watchdog_t)
 ')
+
+ifdef(`init_systemd',`
+	# WRL fix: search /run/systemd/journal (syslogd_var_run_t:s15:c0.c1023)
+	mls_file_read_all_levels(watchdog_t)
+')
-- 
1.9.1

